app/shared/specs/JSONRequest.spec.js

```diff
@@ -303,4 +303,21 @@ describe('JSONRequest', () => {
       });
     });
   });
+
+  describe('head', () => {
+    it('should request the headers for the given url', async () => {
+      const mockresponse = {
+        body: JSON.stringify({ text: 'hey' }),
+      };
+      backend
+        .restore()
+        .head('http://localhost:3000/hello', JSON.stringify({ response: mockresponse }));
+
+      const response = await request.head('http://localhost:3000/hello');
+      const { method } = backend.calls()[0][1];
+
+      expect(method).toBe('HEAD');
+      expect(response.headers).toBeDefined();
```

Comment:
Do you think it would be also useful to assert that there should be not body, which is kind-of the reason behind using head?

Reply:
Well, it's not JSONrequest responsibility to return or not the body, it's part of the HTTP protocol. What I tested is that the correct method is used, and making sure we return the headers if they are present. 

Anyway we are mocking the backend that responses to this calls, and you can actually return a body in a HEAD request, so in order to make it green I just need to dont send a body in the mock, but thats not testing anything from JSONrequest.

Also I could add an if to check the method is "HEAD" and do not return the body, but the point of using HEAD is to avoid large downloads and at that point you already downloaded the body so there is no point in trying to not return it.